File: epi_judge_java_solutions/epi/MaxSumSubarray.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

import java.util.List;

public class MaxSumSubarray {
  @EpiTest(testDataFile = "max_sum_subarray.tsv")

  public static int findMaximumSubarray(List<Integer> A) {

    int maxSeen = 0, maxEnd = 0;
    for (int a : A) {
      maxEnd = Math.max(a, a + maxEnd);
      maxSeen = Math.max(maxSeen, maxEnd);
    }
    return maxSeen;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "MaxSumSubarray.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}

File: epi_judge_python_solutions/max_sum_subarray.py
from typing import List

from test_framework import generic_test


def find_maximum_subarray(A: List[int]) -> int:

    max_seen = max_end = 0
    for a in A:
        max_end = max(a, a + max_end)
        max_seen = max(max_seen, max_end)
    return max_seen


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('max_sum_subarray.py',
                                       'max_sum_subarray.tsv',
                                       find_maximum_subarray))
